src/qibolab/serialize.py

```diff
@@ -34,19 +42,47 @@ def load_qubits(runcard: dict) -> Tuple[QubitMap, QubitPairMap]:
     """
     qubits = {q: Qubit(q, **char) for q, char in runcard["characterization"]["single_qubit"].items()}
 
+    couplers = None
     pairs = {}
-    for pair in runcard["topology"]:
-        pair = tuple(sorted(pair))
-        pairs[pair] = QubitPair(qubits[pair[0]], qubits[pair[1]])
+    if "coupler" in runcard["characterization"]:
+        couplers = {c: Coupler(c, **char) for c, char in runcard["characterization"]["coupler"].items()}
+
+        COUPLER_DISTRIBUTION = list(runcard["topology"].keys())
+        QUBIT_DISTRIBUTION = list(runcard["topology"].values())
+
+        coupler = None
+        for pair in QUBIT_DISTRIBUTION:
+            coupler = couplers[COUPLER_DISTRIBUTION[QUBIT_DISTRIBUTION.index(pair)]]
+            pair = tuple(sorted(pair))
+            pairs[pair] = QubitPair(qubits[pair[0]], qubits[pair[1]], coupler)
```

Comment:
Why not just
```suggestion
        coupler = None
        for c, pair in runcard["topology"].items():
            pair = tuple(sorted(pair))
            pairs[pair] = QubitPair(qubits[pair[0]], qubits[pair[1]], couplers[c])
```
Also, no need to use capitals for local variable names.